app/controllers/support_interface/application_forms/courses_controller.rb

```diff
@@ -0,0 +1,72 @@
+module SupportInterface
+  module ApplicationForms
+    class CoursesController < SupportInterfaceController
+      def new_search
+        @course_search = CourseSearchForm.new
+      end
+
+      def search
+        @course_search       = CourseSearchForm.new(course_search_params)
+        @application_form_id = application_form_id
+
+        if @course_search.valid?
+          redirect_to support_interface_application_form_new_course_path(
+            course_code:         course_search_params[:course_code],
+            application_form_id: @application_form_id,
+          )
+        else
+          render :new_search
+        end
+      end
+
+      def new
+        @pick_course = PickCourseForm.new(
+          course_code: params[:course_code],
+          application_form_id: params[:application_form_id],
+        )
+        @course_code = params[:course_code]
+      end
+
+      def create
+        @course_code = params[:course_code]
+        @pick_course = PickCourseForm.new(
+          course_option_id: course_option_id,
+          course_code: @course_code,
+          application_form_id: params[:application_form_id],
+        )
+
+        if @pick_course.save
+          redirect_to support_interface_application_form_path
+        else
+          render :new
+        end
+      end
+
+    private
+
+      def search_courses
+        Course
+          .current_cycle
+          .includes(course_options: [:site])
+          .where(code: params[:course_code])
```

Comment:
It'd be good to trim leading and trailing whitespace here (quite common for it to get added on touch screen devices). And also upcase to avoid case-sensitivity?

```suggestion
          .where(code: params[:course_code].strip.upcase)
```